### PR TITLE
[sha] Add commit sha in long and short form. Includes test coverage.

### PR DESCRIPTION
**Pull request by scherler**

### CODE REVIEW

File: .fancom [hunk 1]
```diff
@@ -1,41 +0,0 @@
-#     ____
```
Comment (michalbe) [inline .fancom]: Pls. leave this file in the repo.

File: index.js [hunk 2]
```diff
@@ -5,61 +5,88 @@ var exec = require('child_process').exec;
...
-  if (!Array.isArray(gitDataToGet)) {
+  if (gitDataToGet && !Array.isArray(gitDataToGet)) {
     gitDataToGet = [gitDataToGet];
   }
 
+  // this object will be sent to callback, with all the answers
+  const responseObject = {};
+  // We don't want to execute a command that wasn't defined before
+  const filtered = gitDataToGet.filter(filterValid.bind(null, responseObject));
+  if (filtered.length===0) {
```
Comment (michalbe) [inline index.js]: Spaces around `===`

File: package.json [hunk 3]
```diff
@@ -1,8 +1,11 @@
 {
-  "name": "git-info",
-  "version": "1.0.1",
-  "author": "Michał Budzyński <michal@virtualdesign.pl>",
-  "repository": "git@github.com:michalbe/git-info.git",
+  "name": "giti",
+  "version": "1.0.4",
+  "author": "Thorsten Scherler <tscherler@cloudbees.com, scherler@gmail.com>",
+  "contributors": [
+    "Michał Budzyński <michal@virtualdesign.pl>"
```
Comment (michalbe) [inline package.json]: I think it's usually the other way around :).

File: tests/git-info-tests.js [hunk 4]
```diff
@@ -37,5 +43,33 @@ gi('branch', function(err, result) {
...
+  assert.equal(result.shaShort.length, 7);
+}),
+
+gi('sha', function(err, result) {
+  assert(!err);
+  assert('sha' in result);
+  assert.equal(result.sha.length > 7, true);
+});
+
+try {
```
Comment (michalbe) [inline tests/git-info-tests.js]: Use `assert.throws` instead.

**Comment by michalbe** [comment]
@scherler thanks for your contribution!
Also - pls update the `readme` file with new methods you've implemented.

**Comment by michalbe** [comment]
@scherler ping?

**Comment by scherler** [comment]
@michalbe sorry just saw this again. lemme fix that.

**Comment by scherler** [comment]
@michalbe done

**Comment by michalbe** [comment]
@scherler great job, thanks!

**Pull request merged**